src/device-manager/python/openweave/WeaveStack.py

```diff
@@ -0,0 +1,225 @@
+#
+#    Copyright (c) 2020 Google LLC.
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+#
+#    @file
+#      Python interface for Weave Stack
+#
+
+"""Weave Stack interface
+"""
+
+from __future__ import absolute_import
+from __future__ import print_function
+import functools
+import sys
+import os
+import re
+import copy
+import binascii
+import datetime
+import time
+import glob
+import platform
+import ast
+from threading import Thread, Lock, Event
+from ctypes import *
+from .WeaveUtility import WeaveUtility
+
+__all__ = [ 'DeviceStatusStruct', 'WeaveStackException', 'DeviceError', 'WeaveStackError', 'WeaveStack' ]
+
+WeaveStackDLLBaseName = '_WeaveDeviceMgr.so'
+
+def _singleton(cls):
+    instance = [None]
+
+    def wrapper(*args, **kwargs):
+        if instance[0] is None:
+            instance[0] = cls(*args, **kwargs)
+        return instance[0]
+
+    return wrapper
+
+class DeviceStatusStruct(Structure):
+    _fields_ = [
+        ("ProfileId",       c_uint32),
+        ("StatusCode",      c_uint16),
+        ("SysErrorCode",    c_uint32)
+    ]
+
+class WeaveStackException(Exception):
+    pass
+
+class WeaveStackError(WeaveStackException):
+    def __init__(self, err, msg=None):
+        self.err = err
+        if msg != None:
+            self.msg = msg
+        else:
+            self.msg = "Weave Stack Error %ld" % (err)
+
+    def __str__(self):
+        return self.msg
```

Comment:
I think it would be good to cast to str explicitly: return str(self.msg). I ran into cases self.msg was a bytearray

Reply:
Please help reproduce it and share the log, I would fix it in follow-up PR, thanks